server/src/test/java/org/elasticsearch/ingest/IngestServiceTests.java

```diff
@@ -1376,6 +1377,23 @@ public void testResolveFinalPipeline() {
         assertThat(indexRequest.getFinalPipeline(), equalTo("final-pipeline"));
     }
 
+    public void testResolveFinalPipelineWithDateMathExpression() {
+        final DateFormatter dateFormatter = DateFormatter.forPattern("uuuu.MM.dd");
+        IndexMetadata.Builder builder = IndexMetadata.builder("idx-" + dateFormatter.formatMillis(System.currentTimeMillis()))
+            .settings(settings(Version.CURRENT).put(IndexSettings.FINAL_PIPELINE.getKey(), "final-pipeline"))
+            .numberOfShards(1)
+            .numberOfReplicas(0);
+        Metadata metadata = Metadata.builder().put(builder).build();
+
+        // index name matches with IDM:
+        IndexRequest indexRequest = new IndexRequest("<idx-{now/d}>");
+        boolean result = IngestService.resolvePipelines(indexRequest, indexRequest, metadata);
```

Comment:
I believe this will very rarely fail when the first call to `System.currentTimeMillis()` on line 1382 is called on one day, then this is called on the next day (across the day delineation I mean).

Is it worth adding a parameter that allows overriding the time used when resolving for this test? (I only say since we've had failures related to this with data streams and dates in index names)

Reply:
Yep, worth it. Added in https://github.com/elastic/elasticsearch/pull/75314/commits/b934eb4159d120f4a09ce96c8387aa9dcf204750